src/detection/Peak.cc

```diff
@@ -185,9 +141,7 @@ PTR(PeakTable) PeakTable::make(
         }
     }
     // No match: we create a new table and put it in the cache
-    PTR(PeakTable) newTable = std::make_shared<PeakTableImpl>(
-        schema, afw::table::IdFactory::makeSimple()
-    );
+    std::shared_ptr<PeakTable> newTable(new PeakTable(schema, afw::table::IdFactory::makeSimple()));
```

Comment:
The problem is how to declare it as a friend so it can access a private/protected constructor.  I don't think C++11/14 has an answer to that - I think the exact signature you need to friend is not mandated by the standard.  I hope C++17 has a solution, but I haven't been paying attention besides seeing that important people consider it a problem.